node/core/av-store/src/lib.rs

```diff
@@ -478,6 +483,36 @@ impl AvailabilityStoreSubsystem {
 			db,
 			metrics,
 			clock,
+			known_blocks: KnownBlocks::default(),
+			finalized_number: None,
+		}
+	}
+}
+
+#[derive(Default, Debug)]
+struct KnownBlocks {
+	by_hash: HashSet<Hash>,
+	by_number: BTreeSet<(BlockNumber, Hash)>,
+}
```

Comment:
this is kept only in memory, so it might be causing a warn only on startup, but I think it's fine

Reply:
testing my understanding: we're keeping the total set of unfinalized blocks in memory to test whether they're known?

Reply:
perhaps `KnownUnfinalizedBlocks` would communicate this more clearly?

Reply:
yes, we do the same in approval-distribution
> perhaps KnownUnfinalizedBlocks would communicate this more clearly?

:+1: